client/packages/common/src/hooks/useWebClient/useWebClient.ts

```diff
@@ -0,0 +1,20 @@
+import { FileInfo } from './types';
+
+export const useWebClient = () => {
+  const saveFile = (fileInfo: FileInfo) => {
+    const fileData = JSON.stringify({
+      name: fileInfo.filename,
+      content: fileInfo.content,
+    });
+    const blob = new Blob([fileData], { type: 'text/plain' });
+    const url = URL.createObjectURL(blob);
+    const link = document.createElement('a');
+    link.download = fileInfo.filename ?? 'exported-log.txt';
+    link.href = url;
+    link.click();
+  };
+
+  return {
+    saveFile,
+  };
+};
```

Comment:
Maybe unnecessary to separate like this given how it is just one small function. 

I did this to copy existing patterns (as per useNativeClient.ts)

I also wasn't sure if this is the best way or safe way to return save files? This is simple method I found online.